core/block/editor/table/table.go

```diff
@@ -1006,3 +265,38 @@ func (tb Table) Iterate(f func(b simple.Block, pos CellPosition) bool) error {
 	}
 	return nil
 }
+
+// CheckTableBlocksMove checks if Insert operation is allowed in case table blocks are affected
+func CheckTableBlocksMove(st *state.State, target string, pos model.BlockPosition, blockIds []string) (string, model.BlockPosition, error) {
+	// nolint:errcheck
+	if t, _ := NewTable(st, target); t != nil {
+		// we allow moving rows between each other
+		if slice.ContainsAll(t.RowIDs(), append(blockIds, target)...) {
+			if pos == model.Block_Bottom || pos == model.Block_Top {
+				return target, pos, nil
+			}
+			return "", 0, fmt.Errorf("failed to move rows: position should be Top or Bottom, got %s", model.BlockPosition_name[int32(pos)])
+		}
+	}
+
+	for _, id := range blockIds {
+		t := GetTableRootBlock(st, id)
+		if t != nil && t.Model().Id != id {
+			// we should not move table blocks except table root block
+			return "", 0, ErrCannotMoveTableBlocks
+		}
+	}
+
+	t := GetTableRootBlock(st, target)
+	if t != nil && t.Model().Id != target {
+		// we allow inserting blocks into table cell
```

Comment:
we don't allow this, table cell have to be text block without children. But it's a client limitation, so if client devs agreed to support nested block we can support this as well

Reply:
Yes, u r right! Client only are allowed to c/p blocks into table cell, but any way it is being pasted as plaintext.
So in case we are trying to insert some blocks into the table cell, we should insert them under the table